application-development/gae-microservice/client/src/main/java/com/example/appengine/java8/URLFetchClient.java

```diff
@@ -0,0 +1,153 @@
+/**
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Application to demonstrate and test a microservice in App Engine. 
+ */
+package com.example.appengine.java8;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStreamWriter;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.text.DecimalFormat;
+import java.util.logging.Logger;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Servlet connects to a microservice in another project by POSTing
+ * data to a microservice and reading data back using native sockets,
+ * which are supported in App Engine with Java 8. This avoids the use
+ * of the App Engine specific URLFetch API, which was required in
+ * Java 7. For details see 
+ *
+ * https://cloud.google.com/appengine/docs/standard/java/issue-requests
+ */
+@WebServlet(name = "URLFetchClient", value = "/urlfetchtest")
+public class URLFetchClient extends HttpServlet {
+
+  private static final int ITERATIONS = 10;
+  private static final Logger logger = Logger.getLogger(URLFetchClient.class.getName());
+  private static DecimalFormat df = new DecimalFormat(".#");
+
+  @Override
+  public void doGet(HttpServletRequest req, HttpServletResponse res)
+      throws IOException {
+
+    res.setContentType("text/html");
+    logger.info("doGet enter");
+
+    // Setup
+    String filePath =  getServletContext().getRealPath("/static/zeros700k.dat");
+    InputStream is = new FileInputStream(new File(filePath));
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    byte[] buffer = new byte[1024];
+    int length;
+    while ((length = is.read(buffer)) != -1) {
+        baos.write(buffer, 0, length);
+    }
+    String dataToSend = baos.toString("UTF-8");
+
+    // Call service with buffer
+    URL url1 = new URL("https://alexamiesgo.appspot.com/receiveandsend");
```

Comment:
we don't want to hardcode this url right? Perhaps read from form, use current project and fixed service name or environment variable. thanks